app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/LoadFilesListTask.java

```diff
@@ -142,56 +143,91 @@ public LoadFilesListTask(
         if (!hFile.getPath().endsWith("/")) {
           hFile.setPath(hFile.getPath() + "/");
         }
-        try {
-          SmbFile[] smbFile = hFile.getSmbFile(5000).listFiles();
-          list = mainFragment.addToSmb(smbFile, path, showHiddenFiles);
-          openmode = OpenMode.SMB;
-        } catch (SmbAuthException e) {
-          if (!e.getMessage().toLowerCase().contains("denied")) {
+        ArrayList<LayoutElementParcelable> smbCache = mainFragmentViewModel.getFromListCache(path);
```

Comment:
This functions is already really long, can an effort be made to separate it into two?

Reply:
Fixed